driver-core/src/main/java/com/datastax/driver/core/AbstractAddressableByIndexData.java

```diff
@@ -44,42 +44,42 @@ protected ByteBuffer getValue(int i) {
 
     public T setBool(int i, boolean v) {
         checkType(i, DataType.Name.BOOLEAN);
-        return setValue(i, TypeCodec.BooleanCodec.instance.serializeNoBoxing(v));
+        return setValue(i, TypeCodec.booleanCodec.serializeNoBoxing(v));
     }
 
     public T setInt(int i, int v) {
         checkType(i, DataType.Name.INT);
-        return setValue(i, TypeCodec.IntCodec.instance.serializeNoBoxing(v));
+        return setValue(i, TypeCodec.intCodec.serializeNoBoxing(v));
     }
 
     public T setLong(int i, long v) {
         checkType(i, DataType.Name.BIGINT, DataType.Name.COUNTER);
-        return setValue(i, TypeCodec.LongCodec.instance.serializeNoBoxing(v));
+        return setValue(i, TypeCodec.longCodec.serializeNoBoxing(v));
     }
 
     public T setDate(int i, Date v) {
         checkType(i, DataType.Name.TIMESTAMP);
-        return setValue(i, v == null ? null : TypeCodec.DateCodec.instance.serialize(v));
+        return setValue(i, v == null ? null : TypeCodec.dateCodec.serialize(v));
     }
 
     public T setFloat(int i, float v) {
         checkType(i, DataType.Name.FLOAT);
-        return setValue(i, TypeCodec.FloatCodec.instance.serializeNoBoxing(v));
+        return setValue(i, TypeCodec.floatCodec.serializeNoBoxing(v));
     }
 
     public T setDouble(int i, double v) {
         checkType(i, DataType.Name.DOUBLE);
-        return setValue(i, TypeCodec.DoubleCodec.instance.serializeNoBoxing(v));
+        return setValue(i, TypeCodec.doubleCodec.serializeNoBoxing(v));
     }
 
     public T setString(int i, String v) {
         DataType.Name type = checkType(i, DataType.Name.VARCHAR, DataType.Name.TEXT, DataType.Name.ASCII);
         switch (type) {
             case ASCII:
-                return setValue(i, v == null ? null : TypeCodec.StringCodec.asciiInstance.serialize(v));
+                return setValue(i, v == null ? null : TypeCodec.StringCodec.asciiStringCodec.serialize(v));
```

Comment:
You shouldn't access a static field in the parent class through one of its subclasses.